docker/importHistograms.py

```diff
@@ -0,0 +1,136 @@
+import hashlib
+import json
+import sys
+import urllib.request
+
+from google.api_core.exceptions import DeadlineExceeded
+from google.cloud import bigquery, firestore
+
+
+# The number of historical versions to import for each channel.
+NUM_VERSIONS = 10
+# The channels to import data for.
+CHANNELS = ("release", "beta", "nightly")
+
+
+PRODUCT_DETAILS_URL = (
+    "https://product-details.mozilla.org/1.0/firefox_versions.json"
+)
+BQ_TABLE = "moz-fx-data-derived-datasets.telemetry.client_probe_counts_v1"
+BIGQUERY = bigquery.Client()
+FIRESTORE = firestore.Client()
+BATCH_RETRIES = []
+
+
+def commit_batch(batch):
+    try:
+        batch.commit()
+        print(".", end="", flush=True)
+    except DeadlineExceeded:
+        BATCH_RETRIES.append(batch)
+        print("F", end="", flush=True)
+    except:  # noqa
+        print("Unexpected error:", sys.exc_info()[0])
+
+    return FIRESTORE.batch()
+
+
+def import_collection(channel, version):
+
+    query = """
+        SELECT *
+        FROM `{table}`
+        WHERE
+            channel = @channel
+            AND app_version = @app_version
+            AND os IS NOT NULL
+            AND app_build_id IS NOT NULL
+    """.format(
```

Comment:
@emtwo Does this query seem appropriate, with the `IS NOT NULL` clauses?

Reply:
Ah good question, I missed this. The data purposely has rows where `os` and/or `app_build_id` is null because those include *all* values for them. So we probably still need that data if it's possible to have no value set for them in the UI.

Reply:
The prototype didn't allow for selecting the OS or build ID so it's probably best to pull the `NULL` rows in as well. If I understand correctly, that would only add 1 extra row for each per metric, correct?

Reply:
It will add 3 extra rows per metric:
* os=null & build_id is not,
* build_id=null & os is not,
* both are null